### PR TITLE
Added new port learn mode

### PR DESCRIPTION
**Pull request by dgsudharsan**
SAI_PORT_FDB_LEARNING_MODE_FDB_NOTIFICATION mode is proposed so that when a packet from unknown source MAC comes, it is sent as a notification via FDB callback as new MAC learn. This mode will not forward traffic(until application re-installs the FDB entry. This mode will generate only one notification per unknown source MAC to FDB callback.

### CODE REVIEW

File: inc/saiport.h [hunk 1]
```diff
@@ -136,6 +136,9 @@ typedef enum _sai_port_fdb_learning_mode_t
     /** Trap packets with unknown source MAC to CPU. Do not learn. Forward based on destination MAC */
     SAI_PORT_FDB_LEARNING_MODE_CPU_LOG,
 
+    /** Send MAC notification via FDB callback. Do not learn. Do not forward */
```
Comment (AshokDaparthi) [inline inc/saiport.h]: generate FDB Event notifications for unknown source MAC

File: inc/saiport.h [hunk 2]
```diff
@@ -136,6 +136,9 @@ typedef enum _sai_port_fdb_learning_mode_t
...
+    SAI_PORT_FDB_LEARNING_MODE_SW_TRAP,
```
Comment (lguohan) [inline inc/saiport.h]: is it better to change the name to SAI_PORT_FDB_LEARNING_MODE_FDB_NOTIFICATION

Comment (lguohan) [inline inc/saiport.h]: generate notification only for unknown source MAC? Does it generate notification everytime when receiving a same unknown source mac packet repeatedly?

Comment (atitjain) [inline inc/saiport.h]: @lguohan, for this issue to be sorted, one needs to have a FDB entry added as "pending". So on the first packet add that entry to the HW and then let the stack do verification etc and then make the pending entry active.

Comment (dgsudharsan) [inline inc/saiport.h]: This mode will generate only one per unknown source MAC. This is similar to pending entries in hardware where only one notification goes per unknown source MAC

**Comment by AshokDaparthi** [comment]
@itaibaz , @zubinshah @atitjain @tushar-ty @lguohan  - Please review

**Comment by lguohan** [comment]
ok to test

**Comment by AshokDaparthi** [comment]
@itai @mattyk - Could you please let us know support of this behavior.

**Pull request merged**